### PR TITLE
options.cc: document rgw config options

### PR DESCRIPTION
**Pull request by yehudasa**
Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -4604,308 +4760,499 @@ std::vector<Option> get_rgw_options() {
...
     .set_description(""),
 
     Option("rgw_bucket_quota_ttl", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(600)
-    .set_description(""),
+    .set_description("Bucket quota stats cache TTL")
+    .set_long_description(
+        "Length of time for bucket stats to be cached within RGW instance."),
 
-    Option("rgw_bucket_quota_soft_threshold", Option::TYPE_FLOAT, Option::LEVEL_ADVANCED)
```
Comment (mattbenjamin) [inline src/common/options.cc]: the comment should specify the units

Comment (mattbenjamin) [reply src/common/options.cc]: or, rather, that it is a ratio

File: src/common/options.cc [hunk 2]
```diff
@@ -4240,297 +4258,413 @@ std::vector<Option> get_rgw_options() {
...
+    .set_description("RGW FastCGI host name (for FastCGI over TCP)")
+    .add_see_also({"rgw_port", "rgw_fcgi_socket_backlog"}),
 
     Option("rgw_port", Option::TYPE_STR, Option::LEVEL_ADVANCED)
     .set_default("")
-    .set_description(""),
+    .set_description("RGW FastCGI port number (for FastCGI over TCP)")
+    .add_see_also({"rgw_host", "rgw_fcgi_socket_backlog"}),
 
     Option("rgw_dns_name", Option::TYPE_STR, Option::LEVEL_ADVANCED)
```
Comment (theanalyst) [inline src/common/options.cc]: should we drop the level advanced -> level basic in this case?

File: src/common/options.cc [hunk 3]
```diff
@@ -4240,297 +4258,413 @@ std::vector<Option> get_rgw_options() {
...
+    Option("rgw_request_uri", Option::TYPE_STR, Option::LEVEL_DEV)
     .set_default("")
     .set_description(""),
 
     Option("rgw_swift_url", Option::TYPE_STR, Option::LEVEL_ADVANCED)
     .set_default("")
-    .set_description(""),
+    .set_description("Swift-auth storage URL")
+    .set_long_description(
+        "Used in conjuction with rgw internal swift authentication. This affects the "
```
Comment (theanalyst) [inline src/common/options.cc]: conjunction

File: src/common/options.cc [hunk 4]
```diff
@@ -4240,297 +4258,413 @@ std::vector<Option> get_rgw_options() {
...
 
     Option("rgw_swift_enforce_content_length", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(false)
-    .set_description(""),
+    .set_description("Send content length when listing containers (Swift)")
+    .set_long_description(
+        "Whether content length header is needed when listing containers. When this is "
+        "set to false, RGW will send extra info for each entry in the response."),
 
     Option("rgw_keystone_url", Option::TYPE_STR, Option::LEVEL_ADVANCED)
```
Comment (theanalyst) [inline src/common/options.cc]: this also may be downgraded to level basic?

File: src/common/options.cc (hunk 2 above)
Comment (yehudasa) [reply src/common/options.cc]: yeah, although I'm not sure how much we want to make all the fastcgi stuff visible.

**Comment by jcsp** [comment]
Are there still any changes outstanding on this or is it good to go?

**Comment by yehudasa** [comment]
http://pulpito.ceph.com/yehudasa-2017-10-27_11:48:31-rgw-wip-rgw-config-desc-distro-basic-smithi/

**Pull request merged**